shepherd/storage/minio_storage.py

```diff
@@ -298,11 +342,19 @@ def __init__(self, storage_config: StorageConfig):
             if not await self.job_dir_exists(job_id) or not await self._object_exists(job_id, JOB_STATUS_FILE):
                 raise UnknownJobError('Data for job `{}` does not exist'.format(job_id))
 
-            response = await self._client.get_object(Bucket=job_id, Key=JOB_STATUS_FILE)
+            data = BytesIO()
+            await self._get_object(job_id, JOB_STATUS_FILE, data)
         except AioHTTPClientError as he:
             raise StorageInaccessibleError() from he
-        except BotocoreClientError as ce:
+        except StorageError as ce:
             raise StorageError(f"Failed to get status of job `{job_id}`") from ce
 
-        async with response["Body"] as stream:
-            return JobStatusModel(json.loads(await stream.read()))
+        data.seek(0)
+        return JobStatusModel(json.load(data))
+
+    async def close(self) -> None:
+        """
+        Implementation of :py:meth:`shepherd.storage.Storage.close`.
+        """
+
+        await self._session.close()
```

Comment:
If this can raise `AiohttpException` we should catch it and thrown our own exception (and ofc declare it in the interface definition)